byron/chain/executable-spec/src/Cardano/Spec/Chain/STS/Block.hs

```diff
@@ -94,18 +101,27 @@ bUpdPayload b = (b ^. bBody ^. bUpdProp, b ^. bBody ^. bUpdVotes)
 bEndorsment :: Block -> (ProtVer, VKey)
 bEndorsment b = (b ^. bBody ^. bProtVer, b ^. bHeader ^. bhIssuer)
 
-
 -- | Compute the size (in words) that a block takes.
 bSize :: Block -> Natural
-bSize = fromIntegral . abstractSize acctMap
+bSize b = bHeaderSize (b ^. bHeader) + bBodySize (b ^. bBody)
+
+-- | Compute the size (in words) that a block body occupies.
+bBodySize :: BlockBody -> Natural
+bBodySize = fromIntegral . abstractSize costs
   where
-    acctMap = []
+    costs = Map.fromList [ (typeOf (undefined::Maybe UProp), 1)
```

Comment:
Here we're saying that the we assign a constant size to `UProp`. However `UProp` has a `Set` of `STag`s inside:

```haskell
data UProp = UProp
  { _upId :: UpId
  , _upIssuer :: Core.VKey
  , _upParams :: PParams
  , _upPV :: ProtVer
  , _upSwVer :: SwVer
  , _upSig :: Core.Sig UpSD
  , _upSTags :: Set STag
  -- ^ System tags involved in the update proposal.
  , _upMdt :: Metadata
  -- ^ Metadata required for performing software updates.
  } deriving (Eq, Generic, Show)
```

So maybe here we could change this to:

```haskell
(typeOf (undefined::Maybe UProp), 1), (typeOf (undefined::STag), 1)
```
here we're saying: we only care about the number of `STags` inside an update proposal, since this is the only variable data.

Does that make sense?

Reply:
great, I've added a cost (Stag,1) to bBodySize